src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2832,14 +2833,24 @@ template<> void S4Importer::ImportEntity<Guest>(const RCT12SpriteBase& srcBase)
     dst->Angriness = src->angriness;
     dst->TimeLost = src->time_lost;
 
-    for (size_t i = 0; i < 32; i++)
+    std::vector<uint16_t> typesBeenOn;
+    std::vector<ride_id_t> ridesBeenOn;
+    for (uint16_t i = 0; i < RCT2_MAX_RIDE_OBJECTS; i++)
     {
-        dst->RidesBeenOn[i] = src->rides_been_on[i];
+        if (src->ride_types_been_on[i / 8] & (1 << (i % 8)))
+        {
+            typesBeenOn.push_back(i);
+        }
     }
-    for (size_t i = 0; i < 16; i++)
+    for (uint16_t i = 0; i < RCT12_MAX_RIDES_IN_PARK; i++)
     {
-        dst->RideTypesBeenOn[i] = src->ride_types_been_on[i];
+        if (src->rides_been_on[i / 8] & (1 << (i % 8)))
```

Comment:
This needs to be moved into a method, rather than just copypasting it into both of the importer.